cmd/gitops/root/root.go

```diff
@@ -0,0 +1,147 @@
+package root
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"log"
+	"os"
+	"strings"
+
+	runclient "github.com/fluxcd/pkg/runtime/client"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/weaveworks/weave-gitops/cmd/gitops/config"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/utils"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+)
+
+const defaultNamespace = "flux-system"
+
+var (
+	rootCmdFlags      = &config.Options{}
+	kubeconfigArgs    = genericclioptions.NewConfigFlags(false)
+	kubeclientOptions = new(runclient.Options)
+)
```

Comment:
While this is very sleek, I think this won't work easily with enterprise right now, as it's just a copy of this file. I don't think it's insurmountable, but they _will_ be in another package.